architecture/2025-06-compute-extensibility-feature-spec.md

```diff
@@ -0,0 +1,615 @@
+- Provides an avenue for developers to "punch-through" the Radius abstraction, which may be especially necessary for onboarding existing (brownfield) applications on to Radius.
+- Default Recipes can be leveraged for the standard container resource type for platforms that don't require platform-specific configurations. For example, the default Kubernetes Recipe can be used for the standard container resource type without any modifications, while the ACI Recipe can be modified to leverage the platform-specific capabilities.
+
+**Cons:**
+- Exposing the entire set of container configurations in order to allow for platform specific capabilities may be too broad of an abstraction punch-through, resulting in container infrastructure configurations being split or duplicated across two places (i.e. in the container resource definition itself and in the Recipe).
+- May lead to a proliferation of properties in the standard container resource type by developers, making it harder to manage and control for the platform engineering team.
+
+#### Option 2: Not built into the standard container properties
+This option is where the advanced capabilities are not exposed in the standard container properties provided by Radius out of the box, but rather require platform engineers to modify the container resource definitions to enable these capabilities. This could be done by creating a custom Radius Resource Type (RRT) that extends the standard container RRT to include platform-specific properties or configurations.
+
+**Pros:**
+- Keeps the standard container resource type clean and focused on core properties, avoiding clutter from platform-specific configurations or misuse of advanced container configurations by developers.
+- Allows platform engineers to define and enforce specific configurations for advanced capabilities, ensuring consistency across applications deployed on that platform.
+- The Radius maintainers retain the option to add advanced capabilities to the standard container resource type in the future, if needed, without breaking existing applications. For example, if confidential containers become an standard offering across all platforms, the Radius maintainers can add a `confidentialCompute` property to the standard container resource type and provide default Recipes that implement it.
+
+**Cons:**
+- Platform engineers must modify the container resource definitions to enable advanced capabilities, which may require additional effort and coordination.
+- Limits the ability of developers to leverage advanced container features without involving platform engineers, potentially slowing down the development process.
+- Default Recipes may not be registered for the custom container RRT that is created to leverage platform-specific capabilities, requiring platform engineers to create and register custom Recipes for each platform they want to deploy to.
+
```

Comment:
could we publish both to our official definitions?

`Radius.Resources/containers`
`Radius.Resources/extensibleContainers`